### PR TITLE
[Serverless] Fix panic when running the extension without appsec enabled.

### PR DESCRIPTION
**Pull request by purple4reina**
### What does this PR do?

This pull request fixes a bug when running a java function in aws lambda without appsec enabled.

The problem stems from the fact that when a go interface can be implemented by a nil pointer, the interface itself is considered not nil. In the example below, the interface `io.Reader` is implemented in three ways, but only the first of which is considered `nil`. This is despite the second implementation be a nil pointer, as is the case which triggered the panic this PR addresses.

```go
func main() {
	var r io.Reader
	fmt.Println(r == nil)    // true

	r = (*bytes.Buffer)(nil)
	fmt.Println(r == nil)    // false
	
	r = &bytes.Buffer{}
	fmt.Println(r == nil)    // false
}
```

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature. 
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: pkg/serverless/appsec/httpsec/http_test.go [hunk 1]
```diff
@@ -19,58 +17,78 @@ import (
...
 func TestLifecycleSubProcessor(t *testing.T) {
-	t.Setenv("DD_SERVERLESS_APPSEC_ENABLED", "true")
-	asm, err := appsec.New()
-	if err != nil {
-		t.Skipf("appsec disabled: %v", err)
-	}
+	test := func(t *testing.T, appsecEnabled bool) {
+		t.Setenv("DD_SERVERLESS_APPSEC_ENABLED", strconv.FormatBool(appsecEnabled))
+		asm, _, err := appsec.New()
+		require.NoError(t, err)
```
Comment (Julio-Guerra) [inline pkg/serverless/appsec/httpsec/http_test.go:28]:
```suggestion
		if err != nil {
			t.Skipf("appsec disabled: %v", err)
		}
```

Comment (Julio-Guerra) [reply pkg/serverless/appsec/httpsec/http_test.go]: FYI, this is how we have been skipping our tests on our unsupported environments so far.

File: pkg/serverless/appsec/httpsec/http_test.go [hunk 2]
```diff
@@ -86,3 +106,19 @@ func getEventFromFile(filename string) []byte {
...
+
+	assert.True(t, lp.SubProcessor != nil)
+
+	lp.OnInvokeStart(&invocationlifecycle.InvocationStartDetails{
+		InvokeEventRawPayload: []byte(
+			`{"requestcontext":{"stage":"purple"},"httpmethod":"purple","resource":"purple"}`),
+	})
+
+	lp.OnInvokeEnd(&invocationlifecycle.InvocationEndDetails{})
+}
```
Comment (Julio-Guerra) [inline pkg/serverless/appsec/httpsec/http_test.go:124]: I'd suggest you move this test into the lifecycle processor test suite as this is testing its nil-check is working as expected.

**Review by Julio-Guerra — APPROVED** [review]

File: pkg/serverless/appsec/httpsec/http_test.go (hunk 2 above)
Comment (purple4reina) [reply pkg/serverless/appsec/httpsec/http_test.go:124]: Good catch.

**Comment by purple4reina** [comment]
I've determined that the Serverless Integration Tests failures are also occurring on `main` and therefore are not caused by this pull request.

File: cmd/serverless/main.go [hunk 3]
```diff
@@ -276,16 +276,21 @@ func runAgent(stopCh chan struct{}) (serverlessDaemon *daemon.Daemon, err error)
...
 	go func() {
 		defer wg.Done()
-		appsecSubProcessor, appsecProxyProcessor, err = appsec.New()
+		subProcessor, proxySubProcessor, err := appsec.New()
 		if err != nil {
 			log.Error("appsec: could not start: ", err)
 		}
+		if subProcessor != nil {
+			appsecSubProcessor = subProcessor
+		} else if proxySubProcessor != nil {
```
Comment (jcstorms1) [inline cmd/serverless/main.go:291]: Are both `appsecSubProcessor` and `appsecProxyProcessor` supposed to be set? This `else if` will only get called if `subProcessor` is not equal to nil and it wasn't clear if that was intended. I'm guessing we use either a subprocessor or proxy, but not both?

Comment (Julio-Guerra) [reply cmd/serverless/main.go:291]: Yes, you got this right: appsec is either a "proxy subprocessor", or a "regular subprocessor".

**Review by jcstorms1 — APPROVED** [review]

**Pull request merged**